code/obj/item/gun/kinetic.dm

```diff
@@ -853,6 +853,13 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 		set_current_projectile(new/datum/projectile/bullet/tranq_dart)
 		..()
 
+	// Clearer description for invalid dart rifle ammo.
+	attackby(obj/item/ammo/bullets/b, mob/user)
```

Comment:
Uhh noo I meant more like a var/proc that just displays the message and gets called from the parent `attackby`, this is code duplication and will change the behaviour slightly in odd ways.

You could just change the parent `show_text` to `show_text(src.ammo_incompatible_msg)`